src/gpu/intel/ocl/micro_sdpa_configs.hpp

```diff
@@ -32,17 +34,204 @@ struct sdpa_config_t {
+    ukernel_serialized_hwinfo_t(gemmstone::HWInformation &hwInfo)
+        : gmdid(static_cast<uint32_t>(hwInfo.gmdid))
+        , euCount(static_cast<uint32_t>(hwInfo.euCount))
+        , systolicAvailable(hwInfo.systolicAvailable) {}
+
+    uint32_t gmdid;
+    uint32_t euCount;
+    bool systolicAvailable;
+    uint8_t padding[7] = {0};
+};
+DNNL_ASSERT_TRIVIALLY_SERIALIZABLE(ukernel_serialized_hwinfo_t);
+
+struct ukernel_serialized_sizes_t
+    : trivially_serializable_t<ukernel_serialized_sizes_t> {
+
+    ukernel_serialized_sizes_t() = default;
+    ukernel_serialized_sizes_t(gemmstone::SizeParams &sizes)
+        : batch(sizes.batch), m(sizes.m), n(sizes.n), k(sizes.k) {}
+    int64_t batch = 0;
+    int64_t m = 0, n = 0, k = 0;
+};
+DNNL_ASSERT_TRIVIALLY_SERIALIZABLE(ukernel_serialized_sizes_t);
+
+struct ukernel_serialized_problem_t
+    : trivially_serializable_t<ukernel_serialized_problem_t> {
+
+    ukernel_serialized_problem_t() = default;
+    ukernel_serialized_problem_t(gemmstone::GEMMProblem &problem)
+        : Ta_ext(static_cast<uint32_t>(problem.Ta_ext))
+        , Tb_ext(static_cast<uint32_t>(problem.Tb_ext))
+        , Tc_ext(static_cast<uint32_t>(problem.Tc_ext))
+        , Ta(static_cast<uint32_t>(problem.Ta))
+        , Tb(static_cast<uint32_t>(problem.Tb))
+        , Tc(static_cast<uint32_t>(problem.Tc))
+        , Ts(static_cast<uint32_t>(problem.Ts))
+        , A_layout(static_cast<int>(problem.A.layout))
+        , B_layout(static_cast<int>(problem.B.layout))
+        , C_layout(static_cast<int>(problem.C.layout))
+        , B_tileR(problem.B.tileR)
+        , B_tileC(problem.B.tileC)
+        , B_crosspack(problem.B.crosspack)
+        , A_alignment(problem.A.alignment)
+        , A_scale_alignment(problem.A_scale.alignment)
+        , AO_alignment(problem.AO.alignment)
+        , B_alignment(problem.B.alignment)
+        , asPtrDims(problem.asPtrDims)
+        , aOffset(static_cast<int>(problem.aOffset))
+        , Ta_scale(static_cast<uint32_t>(problem.Ta_scale))
+        , A_scale_layout(static_cast<int>(problem.A_scale.layout))
+        , Tao(static_cast<uint32_t>(problem.Tao))
+        , AO_layout(static_cast<int>(problem.AO.layout))
+        , aoPtrDims(problem.aoPtrDims)
+        , aqGroupM(problem.aqGroupM)
+        , aqGroupK(problem.aqGroupK) {}
+
+    uint32_t Ta_ext, Tb_ext, Tc_ext;
+    uint32_t Ta, Tb, Tc, Ts;
+
+    int A_layout;
+    int B_layout;
+    int C_layout;
+
+    uint16_t B_tileR;
+    uint16_t B_tileC;
+    uint8_t B_crosspack;
+
+    uint8_t A_alignment;
+    uint8_t A_scale_alignment;
+    uint8_t AO_alignment;
+    uint8_t B_alignment;
+    uint8_t padding0[7] = {0};
+
+    int asPtrDims;
+    int aOffset;
+
+    uint32_t Ta_scale;
+    int A_scale_layout;
+    uint32_t Tao;
+
+    int AO_layout;
+    int aoPtrDims;
+    int aqGroupM;
+    int aqGroupK;
+
+    bool with_scales, with_zp, with_common_scales;
+    uint8_t padding1[1] = {0};
+};
```

Comment:
Given there's manual padding involved, maybe a static assert for an expected specific size of the class with a comment to update padding as needed for newly added members?